### PR TITLE
ReasonML edition.

### PR DESCRIPTION
**Pull request by fhammerschmidt**
I converted the OCaml edition thanks to the Reason parser which can translate OCaml into Reason source code (and vice versa). Not everything did compile or typecheck, so there was still some manual work necessary.

I figured if I wanted to learn category theory, I'd rather do this in my everyday language so here's the ReasonML edition of CTFP!

### CODE REVIEW

**Comment by hmemcpy** [comment]
Oh wow! This is great, thank you :)
I don't speak ReasonML, so please feel free to tag people you'd think could give a hand in reviewing this! I'd be happy to merge it once you're satisfied with it!

**Comment by fhammerschmidt** [comment]
Of course.

@yawaramin @mlms13 @andywhite37 @Risto-Stevcev
or anybody who feels like it, please review if you want.

File: src/Makefile [hunk 1]
```diff
@@ -36,19 +41,22 @@ OPTFILES = opt-print-ustrade.tex \
...
+all: default scala ocaml reason
 
 scala: suffix='-scala'
 scala: out-dir version.tex $(SCALAPDF)
 
 ocaml: suffix='-ocaml'
 ocaml: out-dir version.tex $(OCAMLPDF)
 
+reason: suffix='-reason'
+reason: out-dir version.tex $(REASONPDF)
```
Comment (hmemcpy) [inline src/Makefile]: due to me being dumb with makefiles, please add a `clean` command here before `out-dir` (see https://github.com/hmemcpy/milewski-ctfp-pdf/commit/78971a95b52e8dc3790194c22b2d7a2e15790955) (also in `all`)

**Comment by andywhite37** [comment]
@fhammerschmidt - this is really cool! I've been meaning to check out this book, so this is a good excuse. I don't know if I'll have time to do a thorough review, but the stuff I saw at a quick glance looked good.

File: src/content/1.1/code/reason/snippet01.re [hunk 2]
```diff
@@ -0,0 +1,6 @@
+module type Polymorphic_Function_F = {
```
Comment (JasoonS) [inline src/content/1.1/code/reason/snippet01.re]: I don't think these module types make it easy to understand if you are not very experienced with reason. I see in Yawar's PR to the ocaml version he also simplified it.

I guess it isn't correct to say `let f: a => b;` alone, but for pedagogical reasons I feel it is better.

Comment (fhammerschmidt) [reply src/content/1.1/code/reason/snippet01.re:1]: I will update accordingly when his PR gets merged.

**Comment by fhammerschmidt** [comment]
Can we please merge this?

It's probably not perfect for now, because I am still waiting for #242 to adapt, but I think it is better to have the bulk merged now.

**Comment by drupol** [comment]
We need to also update the Github actions workflow to include this new edition.

**Comment by hmemcpy** [comment]
Yeah, very sorry for neglecting! :(
Would be happy if you rebased on top of the latest changes to make sure all build workflows are included.
Will be happy to merge then.

**Comment by fhammerschmidt** [comment]
Rebased and added Reason to workflows: https://github.com/hmemcpy/milewski-ctfp-pdf/pull/247/commits/a537ebd9faafcbe753ae746e9ae30b4858f21799

**Comment by hmemcpy** [comment]
Amazing, let's do this! 🎉

**Comment by hmemcpy** [comment]
Thanks all very much for your contributions!!

**Pull request merged**